.storybook/preview.js

```diff
@@ -50,48 +52,83 @@ export const parameters = {
-            <link rel="stylesheet" type="text/css" href="./theming-base-content/content/Base/baseLib/sap_fiori_3_hcb/css_variables.css"></link>
-        `,
-        picked: false
-    },
-    {
-        id: 'HCW_css_variables',
-        code: `
-            <link rel="stylesheet" type="text/css" href="./theming/sap_fiori_3_hcw.css"></link>
-            <link rel="stylesheet" type="text/css" href="./theming-base-content/content/Base/baseLib/sap_fiori_3_hcw/css_variables.css"></link>`,
-        picked: false
+      picked: false
     }
   ],
   docs: {
       container: DocsContainer,
       page: DocsPage,
-      theme: fundamentals
+      theme: fundamentals,
+      transformSource: (src) => {
+        // we strip out the () =>` ` from the story
+        // so that the source can be formatted and
+        // rendered as HTML.
+        const SOURCE_REGEX = /^\(\) => `((.|\n)*)`$/;
+        const match = SOURCE_REGEX.exec(src);
+        return prettify(match ? match[1] : src);
+      },
+  },
+};
+
+// this adds a toolbar addon for switching themes
+export const globalTypes = {
+  theme: {
+    name: 'Theme',
+    description: 'Global theme for components',
+    defaultValue: 'sap_fiori_3',
+    toolbar: {
+      icon: 'paintbrush',
+      items: [
+        { value: 'sap_fiori_3', title: 'Light' },
+        { value: 'sap_fiori_3_dark', title: 'Dark' },
+        { value: 'sap_fiori_3_light_dark', title: 'Light Dark' },
+        { value: 'sap_fiori_3_hcw', title: 'High Contrast Light' },
+        { value: 'sap_fiori_3_hcb', title: 'High Contrast Dark' },
+      ],
+    },
   },
 };
 
+// this decorator is used to inject link style tags
+// based on the current selected theme from the addon toolbar
+// and current component.
+const withThemeProvider = makeDecorator({
+  name: 'withThemeProvider',
+  parameterName: 'themes',
+  wrapper: (storyFn, context, { parameters }) => {
+    let links = [].slice.call(document.getElementsByTagName('link'));
+    let toRemove = [];
+    links.forEach(item => {
+        if(item.attributes['data-theme-id']) {
+            toRemove.push(item);
+        }
+    });
+    let cssArr = context?.parameters?.components || [];
+    cssArr.indexOf('info-label') === -1 && cssArr.push('info-label');
+
+    const styleLinkTag = (stylePath) => {
+      let link = document.createElement('link');
+      link.rel = 'stylesheet';
+      link.href = stylePath;
+      link.setAttribute('data-theme-id', context?.globals?.theme);
+      return link;
+    };
+
+    cssArr.forEach(component => {
+        let stylePath = `${component}-${context?.globals?.theme}.css`;
+        document.head.appendChild(styleLinkTag(stylePath));
+    })
+
+    toRemove.forEach(item => {
+      item.parentNode.removeChild(item);
+    })
+    document.head.append(styleLinkTag(`theming-base-content/content/Base/baseLib/${context?.globals?.theme}/css_variables.css`));
+
+    return storyFn(context);
```

Comment:
This is great! This is what I was trying to do when I added it to the docs page but couldn't figure it out 💯